app/init.php

```diff
@@ -1047,3 +1053,93 @@ function getDevice($root): Device
 
     return $languages;
 });
+
+App::setResource('promiseAdapter', function ($register) {
```

Comment:
I think this is very generic. Let's talk about it in our next sync and mark it as technical debt for now.